packages/manager/apps/pci-databases-analytics/src/pages/services/[serviceId]/namespaces/_components/addEditNamespace.tsx

```diff
@@ -0,0 +1,600 @@
+
+  const { projectId } = useParams();
+  const { form } = useNamespaceForm({
+    editedNamespace,
+    existingNamespaces: namespaces,
+  });
+
+  useEffect(() => {
+    if (!controller.open) form.reset();
+  }, [controller.open]);
+
+  useEffect(() => {
+    if (!isEdition) return;
+    if (showAdvancedConfig) setAdvancedConfiguration(true);
+  }, [showAdvancedConfig]);
+
+  const { t } = useTranslation(
+    'pci-databases-analytics/services/service/namespaces',
+  );
+  const prefix = isEdition ? 'edit' : 'add';
+  const toast = useToast();
+
+  const NamespaceMutationProps: MutateNamespaceProps = {
+    onError(err) {
+      toast.toast({
+        title: t(`${prefix}NamespaceToastErrorTitle`),
+        variant: 'destructive',
+        description: err.response.data.details.message,
+        duration: TOAST.ERROR_DURATION,
+      });
+      if (onError) {
+        onError(err);
+      }
+    },
+    onSuccess(ns) {
+      form.reset();
+      toast.toast({
+        title: t('formNamespaceToastSuccessTitle'),
+        description: t(`${prefix}NamespaceToastSuccessDescription`, {
+          name: ns.name,
+        }),
+      });
+      if (onSuccess) {
+        onSuccess(ns);
+      }
+    },
+  };
+
+  const { addNamespace, isPending: isPendingAddNamespace } = useAddNamespace(
+    NamespaceMutationProps,
+  );
+
+  const { editNamespace, isPending: isPendingEditNamespace } = useEditNamespace(
+    NamespaceMutationProps,
+  );
+
+  const onSubmit = form.handleSubmit((formValues) => {
+    const retentionFormValues: database.m3db.namespace.Retention = {
+      periodDuration: convertDurationStringToISODuration(
+        formValues.periodDuration,
+      ),
+    };
+
+    if (advancedConfiguration) {
+      retentionFormValues.blockDataExpirationDuration = convertDurationStringToISODuration(
+        formValues.blockDataExpirationDuration,
+      );
+
+      retentionFormValues.bufferFutureDuration = convertDurationStringToISODuration(
+        formValues.bufferFutureDuration,
+      );
+
+      retentionFormValues.bufferPastDuration = convertDurationStringToISODuration(
+        formValues.bufferPastDuration,
+      );
+
+      if (!isEdition)
+        retentionFormValues.blockSizeDuration = convertDurationStringToISODuration(
+          formValues.blockSizeDuration,
+        );
+    }
+
+    if (isEdition) {
+      if (Object.entries(form.formState.dirtyFields).length === 0) {
+        onSuccess();
+        return;
+      }
+      const namespace: NamespaceEdition = {
+        id: editedNamespace.id,
+        retention: retentionFormValues,
+        snapshotEnabled: formValues.snapshotEnabled,
+        writesToCommitLogEnabled: formValues.writesToCommitLogEnabled,
+        resolution: convertDurationStringToISODuration(formValues.resolution),
+      };
+
+      editNamespace({
+        projectId,
+        engine: service.engine,
+        serviceId: service.id,
+        namespace,
+      });
```

Comment:
```suggestion
      editNamespace({
        projectId,
        engine: service.engine,
        serviceId: service.id,
        {
          id: editedNamespace.id,
          retention: retentionFormValues,
          snapshotEnabled: formValues.snapshotEnabled,
          writesToCommitLogEnabled: formValues.writesToCommitLogEnabled,
          resolution: convertDurationStringToISODuration(formValues.resolution),
        },
      });
```
It might be easier to read, if we do not use the namespace const anywhere else